tests/main/snap-session-agent-service-control/task.yaml

```diff
@@ -0,0 +1,88 @@
+summary: Check that root can control user instances of systemd via the session agent
+
+systems:
+    # Ubuntu 14.04 does not have a complete systemd implementation
+    - -ubuntu-14.04-*
+    # Ubuntu Core 16 has read-only /etc/systemd/user
+    - -ubuntu-core-16-*
+    # Systemd on CentOS 7/Amazon Linux 2 does not have the user@uid unit
+    - -amazon-linux-2-*
+    - -centos-7-*
+
+environment:
+    TEST_UID: $(id -u test)
+    USER_RUNTIME_DIR: /run/user/${TEST_UID}
+
+prepare: |
+    # Ensure that snapd.session-agent.socket is enabled.  This may not
+    # be the case on distributions where presets have been used to
+    # disable it.
+    if [ ! -L /usr/lib/systemd/user/sockets.target.wants/snapd.session-agent.socket ] &&
+            ! systemctl --user --global is-enabled snapd.session-agent.socket; then
+        systemctl --user --global enable snapd.session-agent.socket
+        touch agent-was-enabled
+    fi
+    mkdir -p "$USER_RUNTIME_DIR"
+    chmod u=rwX,go= "$USER_RUNTIME_DIR"
+    chown test:test "$USER_RUNTIME_DIR"
+    systemctl start "user@${TEST_UID}.service"
+
+    # ensure curl is available (needed for e.g. core18)
+    if ! command -v curl; then
+        snap install --devmode --edge test-snapd-curl
+        snap alias test-snapd-curl.curl curl
+    fi
+
+restore: |
+    if snap list test-snapd-curl; then
+        snap remove test-snapd-curl
+    fi
+    systemctl stop "user@${TEST_UID}.service"
```

Comment:
If the `systemctl stop` command fails, doesn't that mean we are in an unknown state though?  It would mean we've left behind a running user session for the `test` user that could interfere with future tests.  If we somehow failed before starting `user@1234.service`, then the `stop` command is a no-op.

The only places I can see the `systemctl stop ... || true` pattern used in a restore section is for the `proxy` and `proxy-no-core` tests, which dynamically create the unit within the test body.  Here, the unit will always exist.